public/cli/lib/commands/ffmpeg.ts

```diff
@@ -0,0 +1,115 @@
+import fs from 'fs';
+import readdirSorted from 'readdir-sorted';
+import shell from 'shelljs';
+import tempy from 'tempy';
+import path from 'path';
+import { FfmpegSettings } from '../../../models/ffmpegSettings.model';
+
+export async function execute(settings: FfmpegSettings): Promise<void> {
+  // set default value
+  settings.ffmpegPath = settings.ffmpegPath || 'ffmpeg';
+
+  const executeAudioPath = await combineAudioIfNecessary(
+    settings.ffmpegPath,
+    settings.audioFileOrFolderPath,
+    settings.skipAudioFiles
+  );
+
+  return new Promise<void>((resolve, reject) => {
+    shell.exec(
+      `"${settings.ffmpegPath}" -framerate ${settings.framerateIn} -i "${path.join(
+        settings.imagesPath,
+        'frame_%06d.png'
+      )}" -i
+      "${executeAudioPath}" ${settings.framerateOut ? `${settings.framerateOut} ` : ''} -pix_fmt yuv420p "${
+        settings.outputName
+      }"`,
+      (code, stdout, stderr) => {
+        if (code != 0) {
+          const error = new Error(stderr || stdout);
+          reject(error);
+        } else {
+          resolve();
+        }
+      }
+    );
+  });
+}
+
+export async function combineAudioIfNecessary(
+  ffmpegExe: string,
+  fileOrFolderPath: string,
+  skipAudioFiles: Array<string>
+): Promise<string> {
+  return new Promise<string>((resolve, reject) => {
+    // if we have a directory, read the files in the directory
+    if (fs.lstatSync(fileOrFolderPath).isDirectory()) {
+      // read files in the directory
+      readdirSorted(fileOrFolderPath, { numeric: true }).then(async (filesSorted: string[]) => {
+        const files = (filesSorted || []).map((fileName: string) => path.join(fileOrFolderPath, fileName)),
+          mp3Files = files.filter((f: string) => f.indexOf('.mp3') > -1),
+          wavFiles = files.filter((f: string) => f.indexOf('.wav') > -1);
+
+        // If this folder contains wave and mp3 files, then throw error
+        if (mp3Files.length && wavFiles.length) {
```

Comment:
nit: In TS its better to be more explicit about what you are checking thatn what we used to bother with in JS. Eg `if (mp3Files?.length >== 0) {`. Note that the coercion is helpful and alloud in only one place int TS like this `if (mp3Files?.length == null) {`.

Reply:
done.

Reply:
Also in the 2 `else if` below?